store/rootmulti/store.go

```diff
@@ -167,7 +166,32 @@ func (rs *Store) CacheMultiStore() types.CacheMultiStore {
 // Implements interface MultiStore
 // used to createQueryContext, abci_query or grpc query service.
 func (rs *Store) CacheMultiStoreWithVersion(version int64) (types.CacheMultiStore, error) {
-	panic("rootmulti Store don't support historical query service")
+	if version == 0 || (rs.lastCommitInfo != nil && version == rs.lastCommitInfo.Version) {
+		return rs.CacheMultiStore(), nil
+	}
+	opts := rs.opts
+	opts.TargetVersion = uint32(version)
+	opts.ReadOnly = true
+	db, err := memiavl.Load(rs.dir, opts)
```

Comment:
Wouldn't this be very slow or expensive if we need to load everything from disk each time we call this cache multie store function?

Reply:
Yes, this should only be used for infrequent operations, grpc query service should still be handled by versiondb.